apps/gitness/src/framework/context/ThemeContext.tsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect } from 'react'
+
+import { create } from 'zustand'
+import { persist } from 'zustand/middleware'
+
+import { ColorType, ContrastType, FullTheme, ModeType, ThemeState } from '@harnessio/ui/components'
+
+const useThemeStore = create<ThemeState>()(
+  persist(
+    set => ({
+      theme: 'system-std-std' as FullTheme, // Default theme
+      setTheme: (newTheme: FullTheme) => set({ theme: newTheme })
+    }),
+    {
+      name: 'canary-ui-theme' // LocalStorage key
+    }
+  )
+)
+
+export const useTheme = () => {
+  const { theme, setTheme } = useThemeStore(state => ({ theme: state.theme, setTheme: state.setTheme }))
+  return { theme, setTheme }
```

Comment:
Might as well just return it instead of destructering it and then building it back into a new object.
```suggestion
  return useThemeStore(state => ({ theme: state.theme, setTheme: state.setTheme }))
```

Reply:
Done